### PR TITLE
Dselans/protos custom wasm

### PR DESCRIPTION
**Pull request by dselans**
# WIP

@blinktag let's discuss

### CODE REVIEW

File: libs/protos/protos/sp_external.proto [hunk 1]
```diff
@@ -96,6 +97,18 @@ service External {
 
   rpc AppRegisterReject(AppRegisterRejectRequest) returns (StandardResponse);
 
+  // BEGIN Wasm methods
+  rpc GetWasm(GetWasmRequest) returns (GetWasmResponse);
+
+  rpc GetAllWasm(GetAllWasmRequest) returns (GetAllWasmResponse);
+
+  rpc CreateWasm(CreateWasmRequest) returns (StandardResponse);
```
Comment (blinktag) [inline libs/protos/protos/sp_external.proto:105]: Create methods should return a custom response so we can include the ID of the new resource

File: libs/protos/protos/sp_pipeline.proto [hunk 2]
```diff
@@ -117,21 +127,32 @@ message PipelineStep {
     steps.TransformStep transform = 1001;
     steps.EncodeStep encode = 1002;
     steps.DecodeStep decode = 1003;
+
+    // If set, _wasm_id MUST  e filled out and MUST exist
```
Comment (blinktag) [inline libs/protos/protos/sp_pipeline.proto:131]: * MUST be

File: libs/protos/protos/steps/sp_steps_custom.proto [hunk 3]
```diff
@@ -4,7 +4,6 @@ package protos.steps;
 
 option go_package = "github.com/streamdal/streamdal/libs/protos/build/go/protos/steps";
 
-// WIP -- Custom WASM exec?
 message CustomStep {
-  string id = 1;
+  map<string, bytes> args = 1;
```
Comment (blinktag) [inline libs/protos/protos/steps/sp_steps_custom.proto:8]: Don't think we need bytes here. They will be included in `SetPipelinesCommand` and `GetSetPipelinesCommandsByService` messages

Comment (dselans) [reply libs/protos/protos/steps/sp_steps_custom.proto:8]: I was thinking of it as an `interface{}`-like type, but I also just realized that map is dumb as it won't be ordered.

It should be `repeated bytes`, no? Having args as `repeated string` seems incorrect.

**Pull request merged**